### ISSUE TITLE
[Bug] Update-DbaServiceAccount does not update SSL Cert Permz

### ISSUE THREAD

**Issue by potatoqualitee** [sqlcollaborative/dbatools#6230]
If you have a server with an SSL cert ala `Set-DbaNetworkCertificate`, Update-DbaServiceAccount will fail. It needs to do that at some point, just dont know when.

**Comment by niphlod** [comment]
jm2c: if you try and change the service account from the sql server configuration, even "it" doesn't set the relevant perms on the private key, if they were specifically granted to the previous user.
So it's a "bug" in WMI.
TBF, IMHO what usually happens at least in recent mssql setups is that the setup is happy to place the service user where is needed using the well known (but virtual) NT SERVICE\MSSQLSERVER here and there (as in local policies --> perform volume maintenance tasks) and then when someone changes the actual service account it doesn't need to remove the old and add the new ....

I "stole" that idea for the certificates, too. Instead of granting access to the key to the specific service account, I use the virtual account in the ACLs. Sure enough, no more "whoops, I forgot to put the new service account in the private key ACL": and, also, no more "whoops, some OLD service account can still access the private key" because I added the new one in a rush but forgot to remove the old one.


Aside from this trick that may be useful for people running into this issue (use the virtual account on the ACL and forget any further problems), I'd say that Update-DbaServiceAccount "should" grant the read ACL on the cert once it's sure the service account has been changed. Maybe issue a warning that the old ACL can/should be removed, after restarting the service.
